Automattic/wp-calypso
# Editor: Support VideoPress post-editor preview (via Shortcode component)

Supersedes #3832, #2267
Closes #2267

This pull request seeks to implement VideoPress previews in the editor.

![Videos](https://cloud.githubusercontent.com/assets/1779930/18970765/99f3ec52-8660-11e6-9f0c-73a8b657a087.png)

__Implementation Notes:__

The changes in this pull request are nearly identical to those of #3832, where the primary issue was lack of `localStorage` access because of sandboxing of the `<Shortcode />` iframe (necessary for remembering the volume preference). This was remedied by removing the iframe sandboxing, but enforcing that in doing so, the shortcode result be processed by WordPress.com . In the case of VideoPress shortcodes, this should always be accurate since the result will be the same on WordPress.com as it would be on a Jetpack site.

__Testing Instructions:__

Verify that you can insert a VideoPress video in the editor for a WordPress.com or Jetpack site, and that in doing so the video preview is shown in the editor, sized to the videos natural ratio. Next ensure that the video continues to display correctly after saving and refreshing the page, or when toggling between HTML and Visual editing modes.

cc @mjuhasz @timmyc
